wordpress-mobile/WordPress-iOS
# Refactor updating user role to save local user asynchronously

Similar to https://github.com/wordpress-mobile/WordPress-iOS/pull/20239, but this PR only changes the synchronous `performAndSave` to asynchronous one just in one place, since the diff is a bit messy. See pbArwn-5Qy-p2#how-does-this-apply-to-the-wp-and-jp-ios-apps for detailed reasoning.

## Test instructions

Go to My Sites -> People, invite new user and edit their role. Make sure the role is updated correctly.

## Regression Notes
1. Potential unintended areas of impact
None. As the refactored function is only used in the "People" screen.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What described in the "Test instructions"

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
